libdnf/rpm/repo_sack.cpp

```diff
@@ -59,7 +59,11 @@ static void libsolv_repo_free(LibsolvRepo * libsolv_repo) {
 RepoWeakPtr RepoSack::new_repo_from_libsolv_testcase(const std::string & repoid, const std::string & path) {
     std::unique_ptr<std::FILE, decltype(&std::fclose)> testcase_file(solv_xfopen(path.c_str(), "r"), &std::fclose);
     if (!testcase_file) {
-        throw SystemError(EIO, "Unable to open libsolv testcase file", path);
+        try {
+            throw SystemError(errno ? errno : EIO, path);
```

Comment:
You've completely dropped the custom error message ("Unable to open libsolv testcase file"), I don't like this.

Either the constructor has two arguments for a message and a path to the file on which it happened, or you format a string that contains a description as well as the path when throwing the exception...

Reply:
No. This code wraps the system error in an exception. The exception will be caught and the message "Unable to open libsolv testcase file" will be added to the higher-level exception.